### PR TITLE
Merchant order view

### PR DESCRIPTION
**Pull request by vsawchuk**
Create merchant controller action show_order and relevant tests. View allows merchant to only see the products that they own in the order.

### CODE REVIEW

File: app/views/merchants/show_order.html.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  <li>Status: <%= @order.status %></li>
+  <li>Purchase Date: <%= @order.purchase_datetime %></li>
+  <li>Name: <%= @order.cc_name %></li>
+  <li>Email: <%= @order.customer_email %></li>
+  <li>Credit Card: **** **** **** <%= @order.cc_number[-4..-1] %></li>
+  <li>Expiration: <%= @order.cc_expiration %></li>
+  <h4>Mailing Address</h4>
+  <li>
+    <% address_connector = ((@order.address2 == "" || @order.address2 == nil) ? ", " : ", #{@order.address2}, ")%>
+    <%= @order.address1 + address_connector + @order.city + ", " + @order.state + " " +  @order.zipcode %>
```
Comment (skwiens) [inline app/views/merchants/show_order.html.erb:13]: Nice address_connector to format the address!

File: app/views/merchants/show_order.html.erb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+<h3>Purchased Items</h3>
+<% @order.orderitems.each do |item| %>
+  <% if item.product.merchant_id == @merchant.id %>
+    <ul>
+      <li>
+        <%= item.quantity %> <%= link_to item.product.name, product_path(item.product_id) %>: <%= item.product.price * item.quantity %>
+      </li>
+    </ul>
+  <% end %>
+<% end %>
```
Comment (skwiens) [inline app/views/merchants/show_order.html.erb:26]: Would it be helpful to create a method in the merchant model that takes an argument of @order and returns all the orderitems that belong to that merchant to move the if statement out of the view?

**Review by skwiens — APPROVED** [review]
Looks good! Nice work with the traversing through the models to get what you need in a clean way. We still need to add the link to the fullfillment page from the show (merchants) page, but that can be done later.

File: app/views/merchants/show_order.html.erb (hunk 2 above)
Comment (vsawchuk) [reply app/views/merchants/show_order.html.erb:26]: Yes, that is a great idea! I'll make a card for that so we can get to it if we have time.

**Pull request merged**